### PR TITLE
[Skip Issue] Fix some minor bugs in _elementtree.c

### PR DESCRIPTION
**Pull request by ZackerySpytz**

### CODE REVIEW

File: Modules/_elementtree.c [hunk 1]
```diff
@@ -3338,7 +3339,6 @@ _elementtree_XMLParser___init___impl(XMLParserObject *self, PyObject *target,
         if (!target) {
             Py_CLEAR(self->entity);
             Py_CLEAR(self->names);
-            EXPAT(ParserFree)(self->parser);
```
Comment (serhiy-storchaka) [inline Modules/_elementtree.c:3341]: Why this is removed?

Comment (ZackerySpytz) [reply Modules/_elementtree.c:3341]: `self->parser` would be freed again during deallocation, which would cause a crash.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Thank you @ZackerySpytz. Nice patch.

Perhaps some bugs exist in 2.7 too, but sources are too different for automatic merge. Do you mind to make a backport manually if it is necessary?

**Comment by ZackerySpytz** [comment]
I will create a backport to 2.7, Serhiy.

**Pull request merged**